frontend/apps/vara-man/src/feature/tournament-game/Game.tsx

```diff
@@ -18,103 +17,174 @@ import { calculatePoints } from '../game/utils/calculatePoints';
-							<div className="flex gap-3 items-center font-semibold">
-								<Icons.statsTimer />
-
-								<GameTimer isPause={isRegistration || isFinished || gameOver} />
-							</div>
-
-							<div className="flex gap-3 items-center font-semibold">
-								<Icons.statsCoins />
-								{score}
-							</div>
-
-						</div>
-						<div className="flex gap-3 items-center font-semibold cursor-pointer" onClick={() => setGameOver(true)}>
-							<Icons.exit />
-							Exit
-						</div>
-					</div>}
-
-				<GameCanvas isPause={isRegistration || isFinished || !isStarted} />
-
-				<div className="flex gap-5 my-3">
-					<div className="flex gap-3 items-center">
-						<div className="bg-[#DFDFDF] rounded-sm p-1">
-							<ArrowUp color='#767676' />
-						</div>
-
-						<div className="bg-[#DFDFDF] rounded-sm p-1">
-							<ArrowDown color='#767676' />
-						</div>
-
-						<span>Use arrows to move</span>
-					</div>
-
-					<div className="flex gap-3 items-center">
-						<div className="bg-[#DFDFDF] rounded-sm p-1">
-							<ArrowLeft color='#767676' />
-						</div>
-
-						<div className="bg-[#DFDFDF] rounded-sm p-1">
-							<ArrowRight color='#767676' />
-						</div>
-
-						<span>Rotate</span>
-					</div>
-
-					<div className="flex gap-3 items-center">
-						<div className="bg-[#DFDFDF] rounded-sm p-1 px-3 font-bold text-[#726F6F]">
-							Shift
-						</div>
-
-						<span>Hold shift to run</span>
-					</div>
-				</div>
-
-			</div>
-		</div>
-	)
-}
+  const { account } = useAccount();
+  const [isCanceledModal, setCanceledModal] = useState(false);
+  const [activeTab, setActiveTab] = useState('leaderboard');
+  const [playGame, setPlayGame] = useState(false);
+
+  const { tournamentGame, previousGame, setPreviousGame } = useGame();
+
+  const [gameOver, setGameOver] = useAtom(GAME_OVER);
+  const { configState } = useGame();
+  const [coins] = useAtom(COINS);
+
+  const level = tournamentGame?.[0].level || previousGame?.[0].level;
+  const score = configState && calculatePoints(coins, configState, level as IGameLevel);
+
+  const isRegistration = tournamentGame?.[0].stage === 'Registration' || previousGame?.[0].stage === 'Registration';
+  const isFinished = tournamentGame?.[0].stage.Finished || previousGame?.[0].stage.Finished;
+  const isStarted = tournamentGame?.[0].stage.Started || previousGame?.[0].stage.Started;
+
+  useEffect(() => {
+    const admin = tournamentGame?.[0].admin || previousGame?.[0].admin;
+    const isAdmin = admin === account?.decodedAddress;
+
+    if (previousGame && !tournamentGame) {
+      if (!isAdmin) {
+        setCanceledModal(true);
+      } else {
+        setPreviousGame(null);
+      }
+    }
+  }, [tournamentGame]);
+
+  useEffect(() => {
+    if (playGame || isStarted) {
+      setActiveTab('play');
+    }
+  }, [playGame, isStarted]);
+
+  return (
+    <div className="grid gap-4 grid-cols-1 md:grid-cols-3">
+      <div className="tabs-mobile md:hidden">
```

Comment:
It would be more obvious to use `useMediaQuery` as in other places instead of hidden class

Reply:
@vraja-nayaka  useMediaQuery I use only where it is not possible through classes, in other cases it is better to use classes tailwindcss